pkg/asset/manifests/infrastructure.go

```diff
@@ -98,10 +103,21 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 		}
 	case baremetal.Name:
 		config.Status.PlatformStatus.Type = configv1.BareMetalPlatformType
+		// The MAO expects the Provisioining IP in 1.2.3.4/32 CIDR format
+		// But the install-config ClusterProvisioningIP is just the IP
+		provNetIP, provNet, err := net.ParseCIDR(installConfig.Config.Platform.BareMetal.ProvisioningNetworkCIDR)
+		provIP := strings.Replace(provNet.String(), provNetIP, installConfig.Config.Platform.BareMetal.ClusterProvisioningIP
```

Comment:
cc @sadasu let me know if you'd prefer to move this to the MAO

Reply:
I have modified https://github.com/openshift/api/pull/480/files to accept provisioningIP and provisioningNetworkCIDR as 2 seperate config items so that we do not add the CIDR to provisioningIP. As you had mentioned earlier, this is the more intuitive way of getting config.